test/transformer.spec.ts

```diff
@@ -1453,6 +1453,45 @@ describe('transformer', () => {
             expect(ageHidden.hasAttribute('data-setgeopoint')).to.equal(true);
         });
     });
+
+    describe('Optimization regression tests', () => {
+        // These tests are meant to exercise the second condition for definding
+        // `data-form-id`. The XSL seems to suggest that when an explicit `id`
+        // is not present, the value of `xmlns` will be copied instead. That
+        // behavior would be quite odd! Nonetheless, these tests ensure that the
+        // behavior doeesn't change.
+        describe('Potentially surprising behavior: data-form-id', () => {
+            it(`does not copy an explicit xmlns declaration value as a form id when an explicit id is not present`, async () => {
```

Comment:
Wait, so you're saying there's XSL for this but it's wrong and not otherwise tested?